### DISCUSSION TITLE
edit /etc/default/grub for activating GRUB_DISABLE_OS_PROBER

### DISCUSSION THREAD

**Discussion by gaborfri** [Vanilla-OS/live-iso#389] [Q&A]
i have install Vanilla in parallel to Windows, now i  need to activate os-prober in the grub.cfg. 
What is the way to do this?

gabor

**Comment by taukakao** [comment]
`host-shell pkexec nano /etc/default/grub`
and after editing do:
`abroot upgrade -f`

**Reply by gaborfri** [reply]
this was the first step, ic was able to edit the file and also see that os-prober was running,
But as a result there are only two entries in the grub menu, called like before, but the first one run Windows and the second one runs Vanillla Recovery; i have no idea why there are not three entries with the correct names?

gabor

**Reply by taukakao** [reply]
Can't reproduce, it should look like this:

![IMG_20241216_174318](https://github.com/user-attachments/assets/b34f3178-2486-4655-ab6a-0fbb605c197c)

Make sure that your /etc/default/grub has the following entries:
```
GRUB_DEFAULT=0
GRUB_TIMEOUT=5
GRUB_DISABLE_OS_PROBER=false
```

You can change the GRUB_TIMEOUT (it's in seconds) but make sure it's more than 0 if you want to see the proper menu.

**Reply by gaborfri** [reply]
the timeout helps.
what I do not see that there are two menus. the first one show the decision between current and previous state, and the second show the decision between Vanilla and windows.
is it possible to get Windows directly on the first screen?
![IMG_20241216_183547_247](https://github.com/user-attachments/assets/77e74122-c03c-4b95-817e-652c7650994c)
![IMG_20241216_183600_496](https://github.com/user-attachments/assets/402aab4e-24d7-452b-9893-28a288702282)

gabor

**Reply by taukakao** [reply]
That's not possible. The first menu is static to guarantee that Vanilla can always boot.

If this extra menu is too annoying I would recommend using the switcher from your UEFI (if it has one) or to use something like rEFInd
